src/components/Tx/TxHistory/TxEntry.jsx

```diff
@@ -0,0 +1,349 @@
+import React, { Component } from 'react'
+import styled from 'styled-components'
+import PropTypes from 'prop-types'
+import web3 from 'web3'
+import i18n from '../../../i18n'
+import Identicon from '../../Identicon'
+import { networkIdToName, hexToNumberString } from '../../../lib/util'
+
+const BN = web3.utils.BN
+
+export default class TxEntry extends Component {
+  static propTypes = {
+    /** Tx details */
+    tx: PropTypes.shape({
+      nonce: PropTypes.number,
+      from: PropTypes.string.isRequired,
+      to: PropTypes.string,
+      gas: PropTypes.string.isRequired,
+      data: PropTypes.string,
+      gasPrice: PropTypes.string.isRequired,
+      value: PropTypes.string,
+      params: PropTypes.array.isRequired
+    }),
+    /** USD price of ether for displaying tx costs  */
+    etherPriceUSD: PropTypes.number,
+    /** Latest blockNumber */
+    blockNumber: PropTypes.number
+  }
+
+  static linkedTxHash(tx) {
+    if (!tx.hash) {
+      return null
+    }
+
+    let subdomain = ''
+    if (tx.networkId === 3) {
+      subdomain = 'ropsten.'
+    } else if (tx.networkId === 4) {
+      subdomain = 'rinkeby.'
+    } else if (tx.networkId === 42) {
+      subdomain = 'kovan.'
+    }
+
+    return (
+      <a
+        href={`https://${subdomain}etherscan.io/tx/${tx.hash}`}
+        target="_blank"
+        rel="noopener noreferrer"
+      >
+        {tx.hash}
+      </a>
+    )
+  }
+
+  constructor(props) {
+    super(props)
+
+    this.state = { showDetails: false }
+  }
+
+  toggleDetails() {
+    this.setState(prevState => ({ showDetails: !prevState.showDetails }))
+  }
+
+  renderDetails() {
+    const { tx, etherPriceUSD } = this.props
+    const { showDetails } = this.state
+
+    if (!showDetails) {
+      return (
+        <div
+          role="button"
+          tabIndex={0}
+          className="moreDetails"
+          onClick={() => this.toggleDetails()}
+          onKeyDown={() => this.toggleDetails()}
+        >
+          Show details
+        </div>
+      )
+    }
+
+    const linkedTxHash = TxEntry.linkedTxHash(tx)
+    const etherAmount = web3.utils.fromWei(tx.value, 'ether')
+    let etherAmountUSD
+
+    if (tx.networkId === 1 && etherPriceUSD) {
+      etherAmountUSD = (etherAmount * etherPriceUSD).toFixed(2)
+    }
+
+    const gasPriceEther = web3.utils.fromWei(tx.gasPrice, 'ether')
+    const gasPriceGwei = web3.utils.fromWei(tx.gasPrice, 'gwei')
```

Comment:
Ah, we don't have a `weiToGwei` util function yet. Consider adding one for use here, otherwise you'll need to use `weiToEth`, then `ethToGwei`.

Reply:
sounds good